src/rpa_util.F

```diff
@@ -747,6 +756,38 @@ SUBROUTINE calc_fm_mat_S_rpa(fm_mat_S, first_cycle, count_ev_sc_GW, virtual, Eig
 
    END SUBROUTINE calc_fm_mat_S_rpa
 
+! **************************************************************************************************
+!> \brief ...
+!> \param eigen_diff ...
+!> \param Eigenval ...
+!> \param Eigenval_scf ...
+!> \param iocc ...
+!> \param avirt ...
+!> \param homo ...
+!> \param iter_sc_GW0 ...
+! **************************************************************************************************
+   SUBROUTINE get_eigen_diff(eigen_diff, Eigenval, Eigenval_scf, iocc, avirt, homo, iter_sc_GW0)
+      REAL(KIND=dp), INTENT(OUT)                         :: eigen_diff
+      REAL(KIND=dp), DIMENSION(:), INTENT(IN)            :: Eigenval, Eigenval_scf
+      INTEGER, INTENT(IN)                                :: iocc, avirt, homo, iter_sc_GW0
+
+      CHARACTER(LEN=*), PARAMETER :: routineN = 'get_eigen_diff', routineP = moduleN//':'//routineN
+
+      INTEGER                                            :: handle
+
+      CALL timeset(routineN, handle)
+
+      IF (iter_sc_GW0 == 1) THEN
```

Comment:
What about moving this if-clause outside the function (maybe to calc_mat_Q)? Then you have to check that condition only once, and you don't need the argument iter_sc_GW0 and only two of the Eigenval arrays in calc_fm_mat_S_rpa.

Reply:
Thanks, good point, will be changed!